src/lib/courseChannels.js

```diff
@@ -137,4 +142,66 @@ export async function unlinkExistingCourseChannel(roleId, guild) {
     }
 
     await course.destroy();
+}
+
+/**
+ * Updates the colors of the course roles in the server
+ */
+export function updateCourseColors(guild) {
```

Comment:
Is there a reason why you _fully_ avoid using async here? Can't you:

```js
try {
    const courses = await Course.findAll()
    const courseNames = courses.map(course => course.name);
    const courseColors = getCourseColors(courseNames);

    const roleUpdates = courses.map(course => {
        const role = guild.roles.resolve(course.roleId);
        if (!role) {
            reject(`Role with ID ${course.roleId} not found in guild`);
        }
        const color = courseColors[course.name];
        if (!color) {
            reject(`No color found for course ${course.name}`);
        }
        return role.setColor(color);
    });

    await Promise.all(roleUpdates)
        .then(() => `Course colors updated successfully`)
} catch(err) {
    throw `Error fetching courses: ${err}`;
}
```

Reply:
I tried that and it gave unclear error messages. The current method works without any problems.

Reply:
> I tried that and it gave unclear error messages. The current method works without any problems.

Odd, can you send them?